### PR TITLE
v11: Merge `v10/feature/project-cleanup` into `v11/dev`

### PR DESCRIPTION
**Pull request by ronaldbarendse**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

### Description
Merges changes from `v10/feature/project-cleanup` (PR https://github.com/umbraco/Umbraco-CMS/pull/13111, based on the latest `v10/dev`) into `v11/dev` and fixes a lot of merge conflicts because of the project file cleanup done in PR https://github.com/umbraco/Umbraco-CMS/pull/12907.

### CODE REVIEW

File: tests/Umbraco.Tests.Integration/NewBackoffice/OpenAPIContractTest.cs [hunk 1]
```diff
@@ -52,6 +57,6 @@ public async Task Validate_OpenApi_Contract_is_implemented()
             mergedContract.Remove(key);
         }
 
-        Assert.AreEqual(originalGeneratedContract, mergedContract, $"Generated API do not respect the contract:{Environment.NewLine}Expected:{Environment.NewLine}{originalGeneratedContract.ToString(Formatting.Indented)}{Environment.NewLine}{Environment.NewLine}Actual:{Environment.NewLine}{mergedContract.ToString(Formatting.Indented)}");
+        Assert.AreEqual(originalGeneratedContract.ToString(Formatting.Indented), mergedContract.ToString(Formatting.Indented), $"Generated API do not respect the contract.");
```
Comment (nikolajlauridsen) [inline tests/Umbraco.Tests.Integration/NewBackoffice/OpenAPIContractTest.cs:60]: We've had issues where this tests would fail on the pipeline but not locally before, that's why we print the expected/actual, and I think we should keep doing that, since it helps quite a bit when diagnosing

Comment (ronaldbarendse) [reply tests/Umbraco.Tests.Integration/NewBackoffice/OpenAPIContractTest.cs:60]: The test does pass both on the build server and locally now 🤣

And comparing the strings gives a much nicer assertion output:
```
Generated API do not respect the contract.
  Expected string length 137165 but was 137207. Strings differ at index 3445.
  Expected: "...ich are children in relations.",\r\n        "operationId": "..."
  But was:  "...ich are children in relations.\\n            ",\r\n        "o..."
```

Instead of comparing the `JObject`s:
```
Generated API do not respect the contract.
      Expected and actual are both <Newtonsoft.Json.Linq.JObject> with 5 elements
      Values differ at index [2]
        Expected and actual are both <Newtonsoft.Json.Linq.JProperty> with 1 elements
        Values differ at index [0]
          Expected and actual are both <Newtonsoft.Json.Linq.JObject> with 84 elements
          Values differ at index [4]
            Expected and actual are both <Newtonsoft.Json.Linq.JProperty> with 1 elements
            Values differ at index [0]
              Expected and actual are both <Newtonsoft.Json.Linq.JObject> with 1 elements
              Values differ at index [0]
                Expected and actual are both <Newtonsoft.Json.Linq.JProperty> with 1 elements
                Values differ at index [0]
                  Expected and actual are both <Newtonsoft.Json.Linq.JObject> with 6 elements
                  Values differ at index [2]
                    Expected and actual are both <Newtonsoft.Json.Linq.JProperty> with 1 elements
                    Values differ at index [0]
                      Expected and actual are both <Newtonsoft.Json.Linq.JValue>
```

Although, the easiest way to see the difference is by copying the generated API into the `OpenAPI.json` file and doing a git diff (we might want to add that as a comment)... So that might actually make the assertion output less of a problem 🤔

Comment (nikolajlauridsen) [reply tests/Umbraco.Tests.Integration/NewBackoffice/OpenAPIContractTest.cs:60]: Ahh right okay, I guess it's fine then 😄, we just had some weird issues where the OpenApi schema generation differed depending on the build, so we had to compare based on the output from the pipelines 😄 

But I guess that in a pinch we can even compare the two strings, but yeah locally I'd just chuck it in a json diff tool, but the git diff thing is a nice trick as well 😄 

TL;DR ah you're right I think this is fine then 😄

Comment (ronaldbarendse) [reply tests/Umbraco.Tests.Integration/NewBackoffice/OpenAPIContractTest.cs:60]: The previous project settings would only generate XML documentation in Release configuration, so the descriptions weren't available in Debug configuration and caused the schema to differ.

Since that's now fixed and this test can be run in both Debug and Release configuration, this should indeed be fine now! 🥳

Comment (nikolajlauridsen) [reply tests/Umbraco.Tests.Integration/NewBackoffice/OpenAPIContractTest.cs:60]: Ahh amazing, that makes a lot of sense 😄 and it's great that we can fix our schema much easier now 🎉
